sql/core/src/test/scala/org/apache/spark/sql/GeneratorFunctionSuite.scala

```diff
@@ -91,7 +91,7 @@ class GeneratorFunctionSuite extends QueryTest with SharedSQLContext {
     val df = Seq((1, Seq(1, 2, 3)), (2, Seq())).toDF("a", "intList")
     checkAnswer(
       df.select(explode_outer('intList)),
-      Row(1) :: Row(2) :: Row(3) :: Nil)
+      Row(1) :: Row(2) :: Row(3) :: Row(null) :: Nil)
```

Comment:
Based on the definition of `outer`, `outer` has no effect when `join` is false. It should not generate `null`, right?

> @param outer when true, each input row will be output at least once, even if the output of the given `generator` is empty. `outer` has no effect when `join` is false.